OpenRA.Mods.Common/Traits/RepairableNear.cs

```diff
@@ -84,5 +85,17 @@ public void ResolveOrder(Actor self, Order order)
 				self.SetTargetLine(target, Color.Green, false);
 			}
 		}
+
+		public Actor FindRepairBuilding(Actor self)
+		{
+			var repairBuilding = self.World.ActorsWithTrait<RepairsUnits>()
+				.Where(a => !a.Actor.IsDead && a.Actor.IsInWorld
+					&& a.Actor.Owner.IsAlliedWith(self.Owner) &&
+					info.Buildings.Contains(a.Actor.Info.Name))
+				.OrderBy(p => (self.Location - p.Actor.Location).LengthSquared);
+
+			// Worst case FirstOrDefault() will return a TraitPair<null, null>, which is OK.
+			return repairBuilding.FirstOrDefault().Actor;
```

Comment:
repairBuilding is IOrderedEnumerable of TraitPair-s what is Struct, so FirstOrDefault() will never return null, only Actor can be null in such case like the comment says.